src/it/unimi/dsi/fastutil/Threading.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2020 Sebastiano Vigna
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package it.unimi.dsi.fastutil;
+
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.ForkJoinTask;
+
+/**
+ * A utility class for managing how parallel threaded operations are handled in Fastutil.
+ *
+ * <p>For example, this class is how {@link it.unimi.dsi.fastutil.ints#parallelQuickSort} chooses
+ * which {@link ForkJoinPool} to use.
+ *
+ * <p>For methods involving the default pool, the default is only used when a parallel operation is
+ * called and not already in a {@code ForkJoinPool}. If the call already takes place in a
+ * {@code ForkJoinPool}, then that pool is used for subsequent threads regardless of this
+ * default.
+ *
+ * <p>The default default-pool is {@linkplain ForkJoinPool#commonPool() the common ForkJoinPool}.
+ * 
+ * @author C. Sean Young &lt;csyoung@google.com&gt;
+ *
+ */
+public class Threading {
+
+	// We never need to worry about seeing
+	private static volatile ForkJoinPool defaultExecutor = ForkJoinPool.commonPool();
+
+	/**
+	 * Get the pool to use for a Fastutil's parallel operation.
+	 *
+	 * <p>This mimicks the selection logic of {@link java.util.concurrent.ForkJoinTask#fork() ForkJoinTask.fork()},
+	 * except it allows a {@linkplain #getCurrentDefaultFastutilExecutor() custom fallback pool}
+	 * instead of always choosing the {@linkplain ForkJoinPool#commonPool() common pool}. 
+	 * 
+	 * <p>This method is intended for use by Fastutil's classes, not for general use.
+	 * 
+	 * @return the {@linkplain ForkJoinTask#getPool() current pool} if running in a {@link ForkJoinPool},
+	 *   the {@linkplain #getCurrentDefaultFastutilExecutor() current Fastutil default pool} otherwise.
+	 */
+	public static ForkJoinPool getPool() {
+		ForkJoinPool current = ForkJoinTask.getPool();
+		return current == null ? defaultExecutor : current;
+	}
+
+	/**
+	 * Returns the current default executor that will be used by Fastutil's parallel operations not already in a {@link ForkJoinPool}.
+	 */
+	public static ForkJoinPool getCurrentDefaultFastutilExecutor() {
+		return defaultExecutor;
+	}
+
+	/** Set the current default executor that will be used by Fastutil's parallel operations not already in a {@link ForkJoinPool}.
+	 * 
+	 * <p>Note that this is static for the whole program. To override the pool to use for just a
```

Comment:
I'm not sure this is s good idea. If in an application several different methods use fastutil, they could stomp on each other's feet. They could set the default execute at each call, but it seems overkill.

Reply:
Fair enough. Want to just leave the default pool hardcoded to common pool?

Reply:
Yes, I'd just copy-paste where needed something like getPool() != null  ? getPool() : commonPool().

BTW thanks for fixing this, I'm really not an expert on Java parallelism.

Reply:
You mean not even have the Threading class at all? Just inline conditional?

Reply:
Presently to have a class with a single method that is a single conditional seems overkill. I've thought about our places to place the utility methods (e.g., Arrays) but nothing seems to be right. Suggestions are welcome...

Reply:
How about `it.unimi.dsi.fastutil.Arrays`? That is the handwritten, non-typespecific one.
It is a little out of place for a class of that name but currently only Arrays classes have parallel operations.

If not that, it would only be two lines of code to inline into every location. Not that bad.

Reply:
Yeah, I thought of Arrays, but still it's not the right place. It's two lines.